packages/core/src/app/shipping/Shipping.tsx

```diff
@@ -119,38 +133,106 @@ class Shipping extends Component<ShippingProps & WithCheckoutShippingProps, Ship
             deinitializeShippingMethod,
             isMultiShippingMode,
             onToggleMultiShipping,
+            isStripeLinkEnabled,
+            step,
             ...shippingFormProps
         } = this.props;
 
-        const { isInitializing } = this.state;
-
-        return (
-            <div className="checkout-form">
-                <ShippingHeader
-                    isGuest={isGuest}
-                    isMultiShippingMode={isMultiShippingMode}
-                    onMultiShippingChange={this.handleMultiShippingModeSwitch}
-                    shouldShowMultiShipping={shouldShowMultiShipping}
-                />
-
-                <LoadingOverlay isLoading={isInitializing} unmountContentWhenLoading>
-                    <ShippingForm
-                        {...shippingFormProps}
-                        addresses={customer.addresses}
-                        deinitialize={deinitializeShippingMethod}
-                        initialize={initializeShippingMethod}
-                        isBillingSameAsShipping={isBillingSameAsShipping}
-                        isGuest={isGuest}
-                        isMultiShippingMode={isMultiShippingMode}
-                        onMultiShippingSubmit={this.handleMultiShippingSubmit}
-                        onSingleShippingSubmit={this.handleSingleShippingSubmit}
-                        onUseNewAddress={this.handleUseNewAddress}
-                        shouldShowSaveAddress={!isGuest}
-                        updateAddress={updateShippingAddress}
+        const {
+            isInitializing,
+            isStripeLoading,
+            isStripeAutoStep,
+        } = this.state;
+
+        const renderShipping = () => {
+            if (isStripeLinkEnabled && !customer.email) {
+                return <div className="checkout-form">
+                    <div style={ {display: isStripeAutoStep ? 'none' : undefined,} }>
+                    <LoadingOverlay
+                        hideContentWhenLoading
+                        isLoading={ isStripeLoading }
+                    >
+                        <ShippingHeader
+                            isGuest={ isGuest }
+                            isMultiShippingMode={ isMultiShippingMode }
+                            onMultiShippingChange={ this.handleMultiShippingModeSwitch }
+                            shouldShowMultiShipping={ shouldShowMultiShipping }
+                        />
+
+                        <LoadingOverlay
+                            isLoading={ isInitializing }
+                            unmountContentWhenLoading
+                        >
+                            <ShippingForm
```

Comment:
This can be it's own component with props, and can be inside the main component body.